src/App.js

```diff
@@ -0,0 +1,248 @@
+        });
+    }
+
+    function changeList(listName, id) {
+        setListId(id);
+        setPage(listName);
+    }
+
+    function toggleModal() {
+        setShowAlert(!showAlert);
+    }
+
+    function changeModalType(type) {
+        setModalType(type);
+    }
+
+    function updateSharedList(sharedList) {
+        setSharedList(sharedList);
+    }
+
+    return (
+        <div className={"todo"}>
+
+            {page === "home" ?
+                <div>
+                    <h1>TO-DO LIST</h1>
+                    <AddCategory onSubmit={addData}/>
+                    {data.map(item =>
+                        <div>
+                            <ListItem listName={item.listName} onClickItem={changeList}
+                                      handleDeleteList={handleDeleteList} toggleModal={toggleModal}
+                                      id={item.id} showAlert={showAlert}
+                                      setAlertId={setAlertId}
+                                      type={changeModalType}
+                                      sharedWith={item.usersWithAccess}
+                                      isShared={updateSharedList}
+                            />
+
+                        </div>)}
+                </div>
+                :
+                <div>
+                    <h1>{page}</h1>
+                    <TaskList listName={page} db={db} goBack={() => setPage("home")} id={listId}
+                              toggleModal={toggleModal} modalType={"task"} showAlert={showAlert}
+                    />
+                </div>}
+            {page === "home" && showAlert && listExists &&
+            <Alert onClose={toggleModal} onOK={handleEditList} onShare={updateUserswithAccess} type={modalType} id={alertId}
+            sharedWith={sharedList}/>}
+        </div>
+    )
+}
+
+function SignIn() {
+    const [
+        signInWithEmailAndPassword,
+        userCredential, loading, error
+    ] = useSignInWithEmailAndPassword(auth);
+    const [username, setUsername] = useState("");
+    const [password, setPassword] = useState("");
+
+    if (userCredential) {
+        return <div>Unexpectedly signed in already</div>
+    } else if (loading) {
+        return <p>Logging in…</p>
+    }
+    return <div>
+        <div className={"signin"}>
+        {error && <p>"Error logging in: " {error.message}</p>}
+        <input type={"text"} placeholder={"Enter Username"} onChange={(e) => setUsername(e.currentTarget.value)}/>
+        <br/>
+        <input type={"password"} placeholder={"Enter Password"} onChange={(e) => setPassword(e.currentTarget.value)}/>
+        <br/>
+        </div>
+        <div className={"center"}>
+        <button className={"button"} onClick={() =>
+            signInWithEmailAndPassword(username, password)}>Login
+        </button>
+        <button className={"button"} onClick={() =>
+            auth.signInWithPopup(googleProvider)}>Login with Google
```

Comment:
This isn't working: flashes a window that immediately disappears.

Gives console error:  domain is not authorized for OAuth operations for your Firebase project. Edit the list of authorized domains from the Firebase console